neuro_data/movies/stats.py

```diff
@@ -146,3 +149,276 @@ def _make_tuples(self, key):
+        mu = outputs.mean(axis=0, keepdims=True)
+        oracles = ((mu - outputs / r) * r / (r - 1)).reshape(-1, outputs.shape[-1])
+        return oracles
+
+    def sample_and_compute_oracle(self, dataset, condition_hashes, sample_size, min_frames):
+        num_of_neurons = dataset[0].responses.shape[1]
+        
+        # Oracle compuatation
+        true_responses = np.empty(shape=[len(condition_hashes), sample_size * min_frames, num_of_neurons])
+        true_oracles = np.empty(shape=[len(condition_hashes), sample_size * min_frames, num_of_neurons])
+        
+        null_responses = np.empty(shape=[len(condition_hashes), sample_size * min_frames, num_of_neurons])
+        null_oracles = np.empty(shape=[len(condition_hashes), sample_size * min_frames, num_of_neurons])
+        
+        response_matrix = np.empty(shape=[sample_size, min_frames, num_of_neurons]) # this is a temp storage for data extraction
+        
+        for i in range(0, len(condition_hashes)):
+            num_of_neurons = dataset[0].responses.shape[1]
+            dataset_condition_hashes = dataset.condition_hashes
+            # Oracle compuatation
+            true_responses = np.empty(shape=[len(condition_hashes), sample_size * min_frames, num_of_neurons])
+            true_oracles = np.empty(shape=[len(condition_hashes), sample_size *  min_frames, num_of_neurons])
+            
+            null_responses = np.empty(shape=[len(condition_hashes), sample_size * min_frames, num_of_neurons])
+            null_oracles = np.empty(shape=[len(condition_hashes), sample_size *  min_frames, num_of_neurons])
+            
+            response_matrix = np.empty(shape=[sample_size, min_frames, num_of_neurons]) # this is a temp storage for data extraction
+            
+            for i in range(0, len(condition_hashes)):
+                # True Oracle Computation
+                # For each condition_hashes, sample (sample_size) trials to construct the true_response_matrix
+                # Select (sample_size) trials
+                
+                true_target_indices = self.sample_from_condition_hash(condition_hashes[i], dataset_condition_hashes, sample_size)
+                self.check_input(true_target_indices, dataset)
+                
+                
+                for j, index in enumerate(true_target_indices):
+                    response_matrix[j] = self.sample_frames_from_dataset(index, dataset, min_frames, num_of_neurons)
+                    
+                true_responses[i] = response_matrix.reshape(-1, response_matrix.shape[-1])
+                true_oracles[i] = self.compute_oracle(response_matrix)
+                
+                # Null Oracle Computation
+                # Select (samples_size) hashes and sample from them
+                target_hashes = np.random.choice(dataset_condition_hashes, sample_size, replace=False)
+                null_target_indices = np.empty(shape=[target_hashes.size], dtype='int_')
```

Comment:
No need to create an empty array for a sequence of ints.  Lines 328- 332 can be reduced to:

null_target_indices = [self.sample_from_condition_hash(h, dataset_condition_hashes, 1) for h in target_hashes]